packages/core/src/connectors/index.ts

```diff
@@ -1,25 +1,70 @@
+import { AlipayConnector, AlipayConfig } from '@logto/connector-alipay';
+import { AliyunDmConnector, AliyunDmConfig } from '@logto/connector-aliyun-dm';
+import { AliyunSmsConnector, AliyunSmsConfig } from '@logto/connector-aliyun-sms';
+import { FacebookConnector, FacebookConfig } from '@logto/connector-facebook';
+import { GithubConnector, GithubConfig } from '@logto/connector-github';
+import { GoogleConnector, GoogleConfig } from '@logto/connector-google';
+import { SmsConnector, EmailConnector, SocialConnector } from '@logto/connector-types';
+import { WeChatConnector, WeChatConfig } from '@logto/connector-wechat';
+import { WeChatNativeConnector } from '@logto/connector-wechat-native';
+
 import RequestError from '@/errors/RequestError';
 import { findAllConnectors, findConnectorById, insertConnector } from '@/queries/connector';
 
-import * as Alipay from './alipay';
-import * as AliyunDM from './aliyun-dm';
-import * as AliyunSMS from './aliyun-sms';
-import * as Facebook from './facebook';
-import * as GitHub from './github';
-import * as Google from './google';
 import { ConnectorInstance, ConnectorType, IConnector, SocialConnectorInstance } from './types';
-import * as WeChat from './wechat';
-import * as WeChatNative from './wechat-native';
+import { getConnectorConfig, getConnectorRequestTimeout, getFormattedDate } from './utilities';
+
+const getAlipayConfig = async (id: string) => getConnectorConfig<AlipayConfig>(id);
+const AlipayIConnector = new AlipayConnector(
+  getAlipayConfig,
+  getConnectorRequestTimeout,
+  getFormattedDate
+) as SocialConnector;
+
+const getAliyunDmConfig = async (id: string) => getConnectorConfig<AliyunDmConfig>(id);
```

Comment:
We don't need this function, just pass in "getConnectorConfig", the `aliyun-dm` it self can handle the typings by "T".

Reply:
And then, `core` don't need to know the config type of specific connector.